Move torch/lib/c10d to torch/csrc/distributed/c10d

lw opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #60660 Use new canonical path for our own includes of c10d
* **#60543 Move torch/lib/c10d to torch/csrc/distributed/c10d**

Since now c10d is part of libtorch, it would also be nice if the sources lived all in one place.

Differential Revision: [D29062002](https://our.internmc.facebook.com/intern/diff/D29062002/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29062002/)!

lw committed:
Move torch/lib/c10d to torch/csrc/distributed/c10d

Since now c10d is part of libtorch, it would also be nice if the sources lived all in one place.

Differential Revision: [D29062002](https://our.internmc.facebook.com/intern/diff/D29062002/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29062002/)!

[ghstack-poisoned]

lw committed:
Update on "Move torch/lib/c10d to torch/csrc/distributed/c10d"


Since now c10d is part of libtorch, it would also be nice if the sources lived all in one place.

Differential Revision: [D29062002](https://our.internmc.facebook.com/intern/diff/D29062002/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29062002/)!

[ghstack-poisoned]

cbalioglu reviewed (approved):
Overall looks good to me. Hard to spot whether there are any missing items, but since this is mostly a migration work, it should be good to go once all CI tests are green. Approving in advance to unblock you.

lw committed:
Update on "Move torch/lib/c10d to torch/csrc/distributed/c10d"


Since now c10d is part of libtorch, it would also be nice if the sources lived all in one place.

Differential Revision: [D29062002](https://our.internmc.facebook.com/intern/diff/D29062002/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29062002/)!

[ghstack-poisoned]

lw committed:
Update on "Move torch/lib/c10d to torch/csrc/distributed/c10d"


Since now c10d is part of libtorch, it would also be nice if the sources lived all in one place.

Differential Revision: [D29062002](https://our.internmc.facebook.com/intern/diff/D29062002/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29062002/)!

[ghstack-poisoned]

lw committed:
Update on "Move torch/lib/c10d to torch/csrc/distributed/c10d"


Since now c10d is part of libtorch, it would also be nice if the sources lived all in one place.

Differential Revision: [D29062002](https://our.internmc.facebook.com/intern/diff/D29062002/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29062002/)!

[ghstack-poisoned]

lw committed:
Update on "Move torch/lib/c10d to torch/csrc/distributed/c10d"


Since now c10d is part of libtorch, it would also be nice if the sources lived all in one place.

Differential Revision: [D29062002](https://our.internmc.facebook.com/intern/diff/D29062002/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D29062002/)!

[ghstack-poisoned]

malfet reviewed (approved):
👍
